src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -512,6 +515,21 @@ private async Task RestoreAsync(bool forceRestore, RestoreOperationSource restor
                                     {
                                         _status = NuGetOperationStatus.Failed;
                                     }
+
+                                    // Display info bar in SolutionExplorer if there is a vulnerability during restore.
+                                    if (_infoBarService != null)
+                                    {
+                                        bool shouldDisplayVulnerabilitiesInfoBar = AnyProjectHasVulnerablePackageWarning(restoreSummaries);
+                                        if (shouldDisplayVulnerabilitiesInfoBar)
+                                        {
+                                            await _infoBarService.Value.ShowAsync(t);
```

Comment:
If you have a method that takes a `bool`, then this would be easier. 

`_betterServiceName.HasAnyVulnerabilitesInSolution(bool)`.